src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -284,41 +318,52 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
       repairRunner.updateLastEvent(msg);
       return false;
     }
+    
+    boolean gotMetricsForAllHosts = true;
+    boolean connected = false;
 
     for (String hostName : allHosts) {
       LOG.debug("checking host '{}' for pending compactions and other repairs (can repair?)"
                 + " Run id '{}'", hostName, segment.getRunId());
-      try (JmxProxy hostProxy = context.jmxConnectionFactory.connect(hostName)) {
-        int pendingCompactions = hostProxy.getPendingCompactions();
-        if (pendingCompactions > MAX_PENDING_COMPACTIONS) {
-          LOG.info("SegmentRunner declined to repair segment {} because of too many pending "
-                   + "compactions (> {}) on host \"{}\"", segmentId, MAX_PENDING_COMPACTIONS,
-              hostProxy.getHost());
-          String msg = String.format("Postponed due to pending compactions (%d)",
-              pendingCompactions);
-          repairRunner.updateLastEvent(msg);
-          return false;
+      Optional<JmxProxy> hostProxy = Optional.absent();
+      try{
+        Optional<HostMetrics> hostMetrics = Optional.absent();
+        try{
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)